File: javascript/claude/p10_nr_ver2.js
'use strict';

const id_sequence = [10, 20, 30, 40, 50];

function getValueFromArray(arr, size, index) {
    if (!Array.isArray(arr) || size <= 0 || !Number.isInteger(index) || index < 0 || index >= size) {
        return -1;
    }
    return arr[index];
}

function main() {
    const chunks = [];
    process.stdin.setEncoding('utf8');

    process.stdin.on('data', chunk => chunks.push(chunk));

    process.stdin.on('end', () => {
        const raw = chunks.join('').trim();
        const parsed = Number(raw);

        if (!Number.isInteger(parsed) || raw === '') {
            process.exit(-1);
        }

        const result = getValueFromArray(id_sequence, id_sequence.length, parsed);
        process.exit(result);
    });

    process.stdin.on('error', () => {
        process.exit(-1);
    });
}

main();

File: java/claude/p10_nr_ver2.java
import java.util.Scanner;

public class Main {
    static final int[] ID_SEQUENCE = {10, 20, 30, 40, 50};

    static int getValueFromArray(int[] arr, int size, int index) {
        if (arr == null || size <= 0 || index < 0 || index >= size) {
            return -1;
        }
        return arr[index];
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int index;

        try {
            String line = scanner.nextLine().trim();
            index = Integer.parseInt(line);
        } catch (NumberFormatException | java.util.NoSuchElementException e) {
            System.exit(-1);
            return;
        } finally {
            scanner.close();
        }

        int result = getValueFromArray(ID_SEQUENCE, ID_SEQUENCE.length, index);
        System.exit(result);
    }
}